internal/webhook/notebook/mutating.go

```diff
@@ -0,0 +1,310 @@
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/annotations"
+	webhookutils "github.com/opendatahub-io/opendatahub-operator/v2/pkg/webhook"
+)
+
+var (
+	NotebookContainersPath = []string{"spec", "template", "spec", "containers"}
+)
+
+//+kubebuilder:webhook:path=/platform-connection-notebook,mutating=true,failurePolicy=fail,groups=kubeflow.org,resources=notebooks,verbs=create;update,versions=v1,name=connection-notebook.opendatahub.io,sideEffects=None,admissionReviewVersions=v1
+//nolint:lll
+
+// Validator implements webhook.AdmissionHandler for Notebook connection validation webhooks.
+type NotebookWebhook struct {
+	Client  client.Client
+	Decoder admission.Decoder
+	Name    string
+}
+
+// Assert that NotebookWebhook implements admission.Handler interface.
+var _ admission.Handler = &NotebookWebhook{}
+
+func (w *NotebookWebhook) SetupWithManager(mgr ctrl.Manager) error {
+	hookServer := mgr.GetWebhookServer()
+	hookServer.Register("/platform-connection-notebook", &webhook.Admission{
+		Handler:        w,
+		LogConstructor: webhookutils.NewWebhookLogConstructor(w.Name),
+	})
+	return nil
+}
+
+func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
+	log := logf.FromContext(ctx)
+
+	if w.Decoder == nil {
+		log.Error(nil, "Decoder is nil - webhook not properly initialized")
+		return admission.Errored(http.StatusInternalServerError, errors.New("webhook decoder not initialized"))
+	}
+
+	// Validate that we're processing the correct Kind
+	if req.Kind.Kind != gvk.Notebook.Kind {
+		err := fmt.Errorf("unexpected kind: %s", req.Kind.Kind)
+		log.Error(err, "got wrong kind", "group", req.Kind.Group, "version", req.Kind.Version, "kind", req.Kind.Kind)
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	var resp admission.Response
+
+	switch req.Operation {
+	case admissionv1.Create, admissionv1.Update:
+		validationResp, shouldInject, secretRefs := w.validateNotebookConnectionAnnotation(ctx, &req)
+		if !validationResp.Allowed {
+			return validationResp
+		}
+
+		// Only proceed with injection if the annotation is valid and shouldInject is true
+		if !shouldInject {
+			return admission.Allowed(fmt.Sprintf("Connection annotation validation passed in namespace %s for %s, no injection needed", req.Namespace, req.Kind.Kind))
+		}
+
+		injectionPerformed, obj, err := w.performConnectionInjection(req, secretRefs)
```

Comment:
maybe here can already check if `secretRefs` is nil, then skip the `if len(scretRef) == 0` within `performConnectionInjection`